src/config/app.ts

```diff
@@ -67,10 +68,18 @@ export const MEMO_SHARING_SATS_THRESHOLD = yamlConfig.limits.memoSharingSatsThre
 
 export const ONCHAIN_MIN_CONFIRMATIONS = yamlConfig.onChainWallet.minConfirmations
 // how many block are we looking back for getChainTransactions
-export const ONCHAIN_LOOK_BACK = yamlConfig.onChainWallet.lookBack
-export const ONCHAIN_LOOK_BACK_OUTGOING = yamlConfig.onChainWallet.lookBackOutgoing
-export const ONCHAIN_LOOK_BACK_CHANNEL_UPDATE =
-  yamlConfig.onChainWallet.lookBackChannelUpdate
+const getOnChainScanDepth = (val: number): ScanDepth => {
+  const scanDepth = checkedToScanDepth(val)
+  if (scanDepth instanceof Error) throw scanDepth
+  return scanDepth
```

Comment:
shouldn't it be `  return scanDepth as ScanDepth`? or Typescript does it automatically?